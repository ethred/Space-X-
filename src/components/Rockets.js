import { useDispatch, useSelector } from 'react-redux';
import { cancelReservation, reserveRocket } from '../Redux/rockets/rocketsSlice';
import './styles/Rockets.css';

const Rockets = () => {
  const { rockets, status, error } = useSelector((state) => state.rockets);
  const dispatch = useDispatch();
  if (status) {
    return <div className="loading">Loading, please wait...</div>;
  }

  if (error) {
    return (
      <div className="error">Error loading!</div>
    );
  }

  return (
    <div className="rocket_container">
      {rockets.map((rocket) => (
        <section className="space" key={rocket.rocket_id}>
          <div className="img">
            <img src={rocket.flickr_images[0]} alt="" />
          </div>
          <div className="details">
            <div className="headline_h2">{rocket.rocket_name}</div>
            <p className>
              {' '}
              {rocket.reserved ? (
                <span className="reserved">Reserved</span>
              ) : (
                ''
              )}
              {rocket.description}
            </p>
            {rocket.reserved && (
              <button
                type="button"
                className="cancel_reserve_rocket_btn"
                onClick={() => dispatch(cancelReservation(rocket.id))}
              >
                Cancel Reservation
              </button>
            )}
            {!rocket.reserved && (
              <button
                type="button"
                className="reserve_rocket_btn"
                onClick={() => dispatch(reserveRocket(rocket.id))}
              >
                Reserve Rocket
              </button>
            )}
          </div>
        </section>
      ))}
    </div>
  );
};

export default Rockets;
