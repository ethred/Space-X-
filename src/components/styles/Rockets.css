.rocket_container {
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin: 5px 0;
  padding-top: 40px;
  padding-left: 40px;
}

.space {
  display: flex;
  flex-direction: row;
  gap: 10px;
  font-family: sans-serif;
}

.img {
  flex: 1;
}

.img img {
  width: 100%;
}

.details {
  display: flex;
  flex-direction: column;
  flex: 4;
  gap: 10px;
}

.headline_h2 {
  color: #4d5053;
  font-size: 2em;
  font-weight: 600;
  font-family: monospace;
}

.details p {
  color: #4d5053;
  padding-right: 5px;
  font-family: monospace;
  font-size: 1.25rem;
  line-height: 24px;
  padding-left: 10px;
}

.reserved {
  background-color: #68c1d0;
  color: #fff;
  margin-right: 5px;
  border-radius: 5px;
  padding: 5px;
  width: 120px;
  font-size: 1rem;
  font-family: monospace;
}

.reserve_rocket_btn {
  width: 11vw;
  height: 5vh;
  background-color: #007bff;
  border: none;
  color: #fff;
  border-radius: 3px;
  transition: 0.5s;
  font-size: 1rem;
  font-family: monospace;
  cursor: pointer;
}

.reserve_rocket_btn:hover {
  background-color: #094c94;
  color: #000;
}

.cancel_reserve_rocket_btn {
  width: 15vw;
  height: 5vh;
  background-color: #fff;
  border: 1px solid #cdd0d2;
  border-radius: 5px;
  padding: 10px;
  transition: 0.5s;
  color: #373737;
  font-size: 1rem;
  font-family: monospace;
  cursor: pointer;
}
