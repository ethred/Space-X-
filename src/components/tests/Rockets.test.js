import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Rockets from '../Rockets';

const mockStore = configureStore([]);
const initialState = {
  rockets: {
    rockets: [
      {
        rocket_id: '1',
        rocket_name: 'Falcon 9',
        reserved: true,
        description: 'This is a Falcon 9 rocket.',
        flickr_images: ['image_url'],
      },
      {
        rocket_id: '2',
        rocket_name: 'Atlas V',
        reserved: false,
        description: 'This is an Atlas V rocket.',
        flickr_images: ['image_url'],
      },
    ],
    status: 'idle',
    error: null,
  },
};
const store = mockStore(initialState);

test('Rockets component matches snapshot', () => {
  const { container } = render(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
