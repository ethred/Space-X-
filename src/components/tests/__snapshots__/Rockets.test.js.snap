// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component matches snapshot 1`] = `
<div>
  <div
    class="loading"
  >
    Loading, please wait...
  </div>
</div>
`;
